Cargo.toml

```diff
@@ -25,10 +25,11 @@ k256 = { version = "0.11.6", features = ["ecdsa", "std", "pkcs8", "pem"] }
 p256 = { version = "0.11.1", features = ["ecdsa", "std", "pkcs8", "pem"] }
 p384 = { version = "0.11.2", features = ["ecdsa", "std", "pkcs8", "pem"] }
 rand = "0.8.5"
-rsa = "0.6.1"
+rsa = "0.7"
 serde = { version = "1.0.145", features = ["derive"] }
 serde_json = "1.0.86"
-spki05 = { package = "spki", version = "0.5.4" }                           # Use an old version, required by the rsa crate, but that crate doesn't reexport it
+sha2 = { version = "0.10.6", features = ["oid"] }
```

Comment:
Err... now we have more code duplication.

`hmac-sha256` and `hmac-sha512` already provide SHA2 implementations.

Reply:
We could either make your crates depend on sha2, but then they wouldn’t be self-contained anymore, or we could expose stuff in them that satisfy the trait bounds required by the rsa crate. IIRC all the traits come from digest and your crates already depend on it.

Reply:
The current version of the `hmac-*` crates implement the `AssociatedOid` trait, so we shouldn't need `sha2`, just a version bump in `Cargo.toml`.

Reply:
Done :)